src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -854,6 +855,32 @@ public void showReadOnlyWarning(final List<String> alternatives)
       showWarningImpl(() -> warningBar_.showReadOnlyWarning(alternatives));
    }
    
+   @Override
+   public void showRequiredPackagesMissingWarning(List<String> packages)
+   {
+      if (docUpdateSentinel_.hasProperty("disableDependencyDiscovery"))
+      {
+         String disableDependencyDiscovery = docUpdateSentinel_.getProperty("disableDependencyDiscovery");
+         if (StringUtil.equals(disableDependencyDiscovery, "1"))
+         {
+            return;
+         }
+      }
+      
+      CommandWithArg<Boolean> onInstallComplete = (Boolean success) -> {
+         hideWarningBar();
```

Comment:
I think we should keep the warning bar if `success` is false -- that could mean that the user cancelled, or that they couldn't install the packages. In either case we should let them try again.